src/createSchemaForm.tsx

```diff
@@ -397,25 +429,41 @@ export function createTsForm<
       });
     }
     const submitFn = handleSubmit(_submit);
-    type SchemaKey = keyof z.infer<UnwrapEffects<SchemaType>>;
-    const renderedFields = Object.keys(shape).reduce((accum, key : SchemaKey) => {
-      // we know this is a string but TS thinks it can be number and symbol so just in case stringify
-      const stringKey = key.toString();
-      const type = shape[key] as RTFSupportedZodTypes;
+
+
+
+    function renderComponentForSchemaDeep<NestedSchemaType extends RTFSupportedZodTypes | ZodEffects<any, any>, K extends keyof z.infer<UnwrapEffects<SchemaType>> >(
+      _type: NestedSchemaType, props: PropType<Mapping, NestedSchemaType, PropsMapType> | undefined, key: K, prefixedKey: string, currentValue : any
+    ): RenderedElement {
+      const type = unwrapEffects(_type);
       const Component = getComponentForZodType(type, componentMap);
       if (!Component) {
+        if (isAnyZodObject(type)) {
+          const shape: Record<string, RTFSupportedZodTypes> = type._def.shape();
+          return Object.entries(shape).reduce((accum, [subKey, subType]) => {
+            accum[subKey] = renderComponentForSchemaDeep(subType, props && props[subKey] ? (props[subKey] as any) : undefined, subKey, `${prefixedKey}.${subKey}`, currentValue && currentValue[subKey]);
```

Comment:
getting the sub prop type to play nice here was heinous but i can try again when brain is fresh if you feel strongly about this "any"